### DISCUSSION TITLE
Agent & Network

### DISCUSSION THREAD

**Discussion by Thomas-Paul-Fr** [JoelForamitti/agentpy#25] [Q&A] [answered]
Hi there! Thank you again for producing such a nice tool! 

I am a Master's Degree first-year student working on computational models. I am trying to use your virus spread model to apply it to pathogenic molecules in a network. I have "normal" proteins or "misfolded" proteins present in my model.
I have some questions regarding what can be done or not with agentpy:
1) Let's say my network consists of 50 nodes all connected by edges. Can I set a population of agents per node (For instance 50 molecules/node) or can I only have 50 agents? If so, is this then possible to access one specific node to set one specific molecule as "misfolded"?

2) I also wanted to consider the production of the "normal" proteins within time. Is it then possible to add new agents (let's 25 more in every node/ step)?

Most of those questions are related to my understanding of an agent itself I guess. I did not see this topic in the discussion before so I hope it could help others too. Thanks again for this amazing library!

**Comment by jofmi** [comment, marked as answer]
Hi @Thomas-Paul-Fr! Your thesis sounds very interesting! Both of your questions should be possible, but there is no example for it yet in the model library (maybe your model could be added later on if you are interested).

Regarding question 1, if you look at the reference of [`Network.add_agents()`](https://agentpy.readthedocs.io/en/latest/reference_network.html#agentpy.Network.add_agents), you can see that you can add multiple agents to the same AgentNode in the network. You can access these nodes via `Network.nodes`, each being a set from which you can select agents.

Regarding question 2, you can use [`Network.add_agents()`](https://agentpy.readthedocs.io/en/latest/reference_network.html#agentpy.Network.add_agents) at every time-step, so it should be easy to add agents as you go.

**Reply by Thomas-Paul-Fr** [reply]
Hi @JoelForamitti thank you a lot for this reply! So far I am still struggling implementing multiple agents but I will continue today. I'll let you know how it goes!
